### PR TITLE
Unify android and ios setup scripts

### PR DESCRIPTION
**Pull request by catarial**
General cleanup suggested by @shankari

Android and ios builds can be toggled with SETUP_ANDROID and SETUP_IOS environment variables, both are setup if they aren't set.

### CODE REVIEW

**Comment by catarial** [comment]
Oh right I guess this conflicts with our current CI

**Comment by shankari** [comment]
@catarial can you edit the current CI as well? The files are in `.github/workflows`. I want to move from working -> working.

File: setup/setup_native.sh [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    if [ $CURR_RUBY_VERSION == $RUBY_VERSION ]; then
+        echo "Found ruby version "$CURR_RUBY_VERSION" expected "$RUBY_VERSION" no need to upgrade"
+    else
+        if [ -x "${WORKING_DIR}/bin/brew" ]; then
+            echo "Found brew installation with version" ` brew --version`
+            echo "Installing ruby version to brew" $RUBY_VERSION
+            brew install ruby@$RUBY_VERSION
+        else
+            if [ $OSX_MAJOR_VERSION -ge $OSX_EXP_VERSION ]; then
+                echo "No brew installation found, but OSX major version "$OSX_MAJOR_VERSION" and expected version "$OSX_EXP_VERSION" so CocoaPods should work"
```
Comment (shankari) [inline setup/setup_native.sh:42]: nit; future fix: I think we should remove this check. Per @iantei, it doesn't actually work on recent versions of android. We should just say that brew is a requirement.

File: setup/setup_native.sh [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+            else
+                echo "No brew installation found, but OSX major version "$OSX_MAJOR_VERSION" != expected version "$OSX_EXP_VERSION" CocoaPods install will likely fail"
```
Comment (shankari) [inline setup/setup_native.sh:44]: nit, future fix: we might as well remove this as well, if we are going to make `brew` required

**Review by shankari — APPROVED** [review]
LGTM! I have some nits/suggestions for future polishing

**Pull request merged**